### ISSUE TITLE
`last` link occasionally dead leading to sessions not restored

### ISSUE THREAD

**Issue by valentinrusu-movidius** [tmux-plugins/tmux-resurrect#294]
Hey, I already saw closed #93 issue, but in my case the problem still produces from time to time. For instance, today.

$ cd ~/.tmux/resurrect  
$ ll last
lrwxrwxrwx last -> tmux_resurrect_20190318T105819.txt

$ ll tmux_resurrect_20190318T105819.txt
ls: cannot access 'tmux_resurrect_20190318T105819.txt': No such file or directory

Then I unlink last then re-create it to point to an existing file and then it works.

**Comment by Blake-LeBlanc** [comment]
Sounds like the same thing I was experiencing here: #287 

I was never able to fix the issue. Until this gets resolved, I've since started using something called [tmuxp](https://github.com/tmux-python/tmuxp), which also helps manage session layouts.

**Comment by valentinrusu-movidius** [comment]
> I was never able to fix the issue. Until this gets resolved, I've since started using something called [tmuxp](https://github.com/tmux-python/tmuxp), which also helps manage session layouts.

Well, I'll try to figure it out myself, when I'll have some time, as I prefer this shell-only plugin over tmuxp.

**Comment by v9v** [comment]
I've noticed this after I started running two tmux sessions simultaneously. Also sometimes I get duplicate lines in resurrect files (https://github.com/tmux-plugins/tmux-continuum/issues/3).

It looks like sometimes there are multiple saves running in parallel. That explains duplicate lines in the resurrect file, and also occasional removal of the file right after the creation caused by a race condition in the following code in save.sh/save_all():

```
        if files_differ "$resurrect_file_path" "$last_resurrect_file"; then      
                ln -fs "$(basename "$resurrect_file_path")" "$last_resurrect_file"
        else                                                                     
                rm "$resurrect_file_path"                                        
        fi                                                                       

```

If processes A and B started saving in the same second, then they have the same $resurrect_file_path. Then there is a change that process A executes that code fragment first and creates the link, and a moment later process B sees that there is no diff between $resurrect_file_path and "last" (because last is pointing to $resurrect_file_path) and deletes the file.

As a quick workaround, I appended PID to $resurrect_file_path. After a while, I've noticed that ~/.tmux/resurrect has *two* files created in the same second by different processes:

tmux_resurrect_2019-07-04T11:11:26_59803.txt
tmux_resurrect_2019-07-04T11:11:26_59815.txt

which confirms the theory above. The contents of the files is identical.
I haven't found a way to reliably reproduce that though.
